### PR TITLE
WIP: NFT architecture document

### LINKED ISSUE #103: Design NFT architecture with examples from BNS
Start from Isabella's design and discussion with Ethan and Alex (to be pushed as link and wip pr)

Work through the details and consider the three example NFTs we will need for BNS, considering what common functionality we can abstract out, and what needs to be custom for each NFT.

For this step, we consider creating an NFT involved writing new code and deploying a new version of the blockchain, but we can create a package with common functionality that we can import from each example, to make coding a new NFT fast and easy.

Later iterations can consider the possibility of no-code deployments, but this is moving into the realm of VM and smart contracts which are out of scope for launch.

This informs the NFT implementation tasks, and the docs should also be updated from the learnings there (full feedback loop).

### PR DESCRIPTION
**Pull request by ethanfrey**
This is a beginning designing what we want with NFTs.
Please expand on this and complete it to a point that it can inform development.

Resolves #103

### CODE REVIEW

**Comment by ethanfrey** [comment]
@alpe @ruseinov if you like this, please merge. Or feel free to edit the text in the PR (I know there were several questions on slack, just surprised no movement here)

File: architecture/NFT.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+`timeout: int` ? : The approval may automatically expire at some block height (or timestamp?). This will not trigger a cron job, but usage after the timeout will just error (and trigger a cleanup maybe?)
+
+`immutible: boolean` ? : If ownership of the NFT is transfered,
+it may be normal for all Approvals to be revoked, so the new owner
+can re-issue them as needed. If I buy your domain name, I certainly
+don't expect your sysadmins to be able to set my DNS records.
+However, there are cases where the approval will persist beyond
+a transfer of ownership (like a lease of mineral rights to some land).
+The immutible flag could be used to renote the approval does not
+expire upon transfer of ownership.
```
Comment (alpe) [inline architecture/NFT.md]: Q: How do we want to handle a combination of options. Immutible but timeout expired or immutible but count =0 ?

Comment (ruseinov) [reply architecture/NFT.md]: A nitpick: should that be immutable instead?

Comment (ethanfrey) [reply architecture/NFT.md]: Yes, please add to the docs and push to this branch. With the spelling fix, but also with the clairifications that arise from your spike.

Or copy this doc into your spike and edit there if you intend to merge the spike later. In any case, this doc should make it into master when it is refined.

**Pull request merged**